### PR TITLE
[WASM] Initial code drop for debugger

### PR DESCRIPTION
**Pull request by kumpera**
This supports setting, resolving and hitting breakpoints.

This support reporting stack frames and their params & locals.

### CODE REVIEW

**Comment by kumpera** [comment]
The two failures are unrelated.

File: mono/metadata/object-offsets.h [hunk 1]
```diff
@@ -183,7 +183,6 @@ DECL_OFFSET(MonoContext, wasm_bp)
 DECL_OFFSET(MonoContext, wasm_sp)
 DECL_OFFSET(MonoContext, llvm_exc_reg)
 
-DECL_OFFSET(MonoLMF, method)
```
Comment (lewurm) [inline mono/metadata/object-offsets.h]: why was this removed? this breaks at least ARM (it likely others): https://github.com/mono/mono/blob/99198da52bee6d5a648a3e92923693bf56cb4d18/mono/mini/tramp-arm.c#L245

Comment (lewurm) [reply mono/metadata/object-offsets.h]: nevermind, I didn't see the `ifdef` above.
